client/apps/edit2/components/content/section_controls/index.jsx

```diff
@@ -34,143 +44,122 @@ export class SectionControls extends Component {
-        <a
-          name='column_width'
-          className='layout'
-          onClick={() => this.changeLayout('column_width')}
-          data-active={section.get('layout') === 'column_width'} />
-        {
-          this.sectionHasFullscreen(section) &&
-          <a
-            name='fillwidth'
-            className='layout'
-            onClick={() => this.changeLayout('fillwidth')}
-            data-active={section.get('layout') === 'fillwidth'}>
-            <IconImageFullscreen fill={'white'} />
-          </a>
-        }
-        {
-          this.hasImageSet() &&
-          <a
-            name='image_set'
-            className='layout'
-            onClick={this.toggleImageSet}
-            data-active={section.get('type') === 'image_set'} />
+  insideComponent = () => {
+    const { isHero } = this.props
+    if (this.controls) {
+      const $section = $(this.controls).closest('section')
+
+      let insideComponent = false
+      const isScrollingOver = this.isScrollingOver($section)
+      const isScrolledPast = this.isScrolledPast($section)
+
+      if ($section && !isHero) {
+        if (
+          (isScrollingOver && !isScrolledPast) ||
+          (isHero && !isScrollingOver)
+        ) {
+          insideComponent = true
         }
-      </nav>
-    )
+      }
+      return insideComponent
+    }
   }
 
   render () {
-    const { isHero, section, sectionLayouts } = this.props
+    const {
+      children,
+      disabledAlert,
+      isHero,
+      showLayouts
+    } = this.props
     const { insideComponent } = this.state
-    const isSticky = insideComponent ? ' sticky' : ''
+
     const outsidePosition = isHero ? 'relative' : 'absolute'
+    const position = insideComponent ? 'fixed' : outsidePosition
+    const bottom = this.getPositionBottom()
 
     return (
-      <div
-        ref='controls'
-        className={'edit-controls' + isSticky}
-        style={{
-          position: insideComponent ? 'fixed' : outsidePosition,
-          bottom: this.getPositionBottom()
+      <SectionControlsContainer
+        innerRef={(node) => { this.controls = node }}
+        className={'SectionControls edit-controls'}
+        position={position}
+        bottom={bottom}
+      >
+
+        {showLayouts &&
+          <LayoutControls
+            disabledAlert={disabledAlert}
+          />
         }
-      }>
-        { sectionLayouts &&
-          this.renderSectionLayouts(section) }
+
         <div className='edit-controls__inputs'>
-          {this.props.children}
+          {children}
         </div>
-      </div>
+
+      </SectionControlsContainer>
     )
   }
 }
 const mapStateToProps = (state) => ({
-  article: state.edit.article,
   channel: state.app.channel
 })
 
 export default connect(
   mapStateToProps
 )(SectionControls)
+
+const SectionControlsContainer = styled.div`
+  bottom: ${props => props.bottom};
+  position: ${props => props.position};
+`
```

Comment:
I think it's ~ ok to start mixing in styled-components with old CSS but I would also be wary of creeping drift as this kind of thing is typically refactored out all at once (in phases) and can get increasingly unmaintainable otherwise. Def don't want future devs adding stuff to both systems when a single system can be used.

Reply:
This comes from ongoing conversation with @kanaabe -- rather than using inline styles to pass style-related props we have been trying to transition to StyledComponents.

Reply:
Sounds good 👍, just something to be cautious about and not abuse.